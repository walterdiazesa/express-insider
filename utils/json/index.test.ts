import { colorizedJSON, getCircularReplacer } from ".";

const mockExpenses = [
	{
		"id": "debd3d01-3dd5-6edd-9593-7166c8706e04",
		"title": "Spar",
		"purchaseDate": "2023-05-02T11:40:00.000Z",
		"type": "minus",
		"from": "CASH",
		"currency": "USD",
		"amount": "10",
		"orCurrency": null,
		"orAmount": null,
		"owner": "walterwalon@gmail.com",
		"categoryId": 2,
		"category": {
			"id": 2,
			"name": "ðŸ¥– Alimentos",
			"color": "F9CE58"
		}
	},
	{
		"id": "debd3d01-3dd5-6edd-9593-7166c8706e04",
		"title": "Spar",
		"purchaseDate": "2023-05-02T11:40:00.000Z",
		"type": "minus",
		"from": "CASH",
		"currency": "USD",
		"amount": "10",
		"orCurrency": null,
		"orAmount": null,
		"owner": "walterwalon@gmail.com",
		"categoryId": 2,
		"category": {
			"id": 2,
			"name": "ðŸ¥– Alimentos",
			"color": "F9CE58"
		}
	},
	{
		"id": "debd3d01-3dd5-6edd-9593-7166c8706e04",
		"title": "Spar",
		"purchaseDate": "2023-05-02T11:40:00.000Z",
		"type": "minus",
		"from": "CASH",
		"currency": "USD",
		"amount": "10",
		"orCurrency": null,
		"orAmount": null,
		"owner": "walterwalon@gmail.com",
		"categoryId": 2,
		"category": {
			"id": 2,
			"name": "ðŸ¥– Alimentos",
			"color": "F9CE58"
		}
	}
] as const;

describe('colorizedJSON', () => {
  it('should print mockExpenses with corresponding colors', () => {
    expect(colorizedJSON('    ', mockExpenses)).toMatchSnapshot()
  })
  it('should print object without property "n" with corresponding colors and single line', () => {
    expect(colorizedJSON('    ', { query: { title: 'spar', from: '20202', n: undefined }, params: { only: 2 } }, [,false])).toMatchSnapshot()
  })
  it('should print object without property "n" with corresponding colors and multi line', () => {
    expect(colorizedJSON('    ', { query: { title: 'spar', from: '20202', n: undefined }, params: { only: 2 } }, [,true])).toMatchSnapshot()
  })
	it('Should print object without circular reference and with cyan function', () => {
		const req = {
			url: '/',
			rawHeaders: ['http://localhost:3000/', 'application/json']
		};
		(req as any).res = {
			send: (value: any) => {},
			req
		}
		expect(colorizedJSON('', JSON.parse(JSON.stringify(req, getCircularReplacer())))).toMatchSnapshot();
	})
});

describe('circularReferenceReplacer', () => {
	it('Should remove the circular reference', () => {
		const req = {
			url: '/',
			rawHeaders: ['http://localhost:3000/', 'application/json']
		};
		(req as any).res = {
			send: (value: any) => {},
			req
		}
		expect(() => JSON.stringify(req)).toThrow();
		expect(JSON.parse(JSON.stringify(req, getCircularReplacer()))).toEqual({
			...req,
			res: {
				send: {
					type: "[*Function*]ref",
					name: "send",
				}
			}
		})
	})
})