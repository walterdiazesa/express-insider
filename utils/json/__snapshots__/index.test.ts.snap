// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colorizedJSON Should print object without circular reference and with cyan function 1`] = `
"{
  [33murl: [0m[0m[32m"/"[0m,
  [33mrawHeaders: [0m[
    [0m[32m"http://localhost:3000/"[0m,
    [0m[32m"application/json"[0m
  ],
  [33mres: [0m{
    [33msend: [0m[36m[Function send][0m
  }
}"
`;

exports[`colorizedJSON should print mockExpenses with corresponding colors 1`] = `
"[
      {
        [33mid: [0m[0m[32m"debd3d01-3dd5-6edd-9593-7166c8706e04"[0m,
        [33mtitle: [0m[0m[32m"Spar"[0m,
        [33mpurchaseDate: [0m[0m[32m"2023-05-02T11:40:00.000Z"[0m,
        [33mtype: [0m[0m[32m"minus"[0m,
        [33mfrom: [0m[0m[32m"CASH"[0m,
        [33mcurrency: [0m[0m[32m"USD"[0m,
        [33mamount: [0m[0m[32m"10"[0m,
        [33morCurrency: [0m[0m[31mnull[0m,
        [33morAmount: [0m[0m[31mnull[0m,
        [33mowner: [0m[0m[32m"walterwalon@gmail.com"[0m,
        [33mcategoryId: [0m[0m[34m2[0m,
        [33mcategory: [0m{
          [33mid: [0m[0m[34m2[0m,
          [33mname: [0m[0m[32m"ðŸ¥– Alimentos"[0m,
          [33mcolor: [0m[0m[32m"F9CE58"[0m
        }
      },
      {
        [33mid: [0m[0m[32m"debd3d01-3dd5-6edd-9593-7166c8706e04"[0m,
        [33mtitle: [0m[0m[32m"Spar"[0m,
        [33mpurchaseDate: [0m[0m[32m"2023-05-02T11:40:00.000Z"[0m,
        [33mtype: [0m[0m[32m"minus"[0m,
        [33mfrom: [0m[0m[32m"CASH"[0m,
        [33mcurrency: [0m[0m[32m"USD"[0m,
        [33mamount: [0m[0m[32m"10"[0m,
        [33morCurrency: [0m[0m[31mnull[0m,
        [33morAmount: [0m[0m[31mnull[0m,
        [33mowner: [0m[0m[32m"walterwalon@gmail.com"[0m,
        [33mcategoryId: [0m[0m[34m2[0m,
        [33mcategory: [0m{
          [33mid: [0m[0m[34m2[0m,
          [33mname: [0m[0m[32m"ðŸ¥– Alimentos"[0m,
          [33mcolor: [0m[0m[32m"F9CE58"[0m
        }
      },
      {
        [33mid: [0m[0m[32m"debd3d01-3dd5-6edd-9593-7166c8706e04"[0m,
        [33mtitle: [0m[0m[32m"Spar"[0m,
        [33mpurchaseDate: [0m[0m[32m"2023-05-02T11:40:00.000Z"[0m,
        [33mtype: [0m[0m[32m"minus"[0m,
        [33mfrom: [0m[0m[32m"CASH"[0m,
        [33mcurrency: [0m[0m[32m"USD"[0m,
        [33mamount: [0m[0m[32m"10"[0m,
        [33morCurrency: [0m[0m[31mnull[0m,
        [33morAmount: [0m[0m[31mnull[0m,
        [33mowner: [0m[0m[32m"walterwalon@gmail.com"[0m,
        [33mcategoryId: [0m[0m[34m2[0m,
        [33mcategory: [0m{
          [33mid: [0m[0m[34m2[0m,
          [33mname: [0m[0m[32m"ðŸ¥– Alimentos"[0m,
          [33mcolor: [0m[0m[32m"F9CE58"[0m
        }
      }
    ]"
`;

exports[`colorizedJSON should print object without property "n" with corresponding colors and multi line 1`] = `
"{
      [33mquery: [0m{
        [33mtitle: [0m[0m[32m"spar"[0m,
        [33mfrom: [0m[0m[32m"20202"[0m
      },
      [33mparams: [0m{
        [33monly: [0m[0m[34m2[0m
      }
    }"
`;

exports[`colorizedJSON should print object without property "n" with corresponding colors and single line 1`] = `
"{
      [33mquery: [0m{ [33m"title": [0m[32m"spar"[0m, [33m"from": [0m[32m"20202"[0m },
      [33mparams: [0m{ [33m"only": [0m[34m2[0m }
    }"
`;
